internal/handlers/pg/pgdb/query_test.go

```diff
@@ -168,3 +172,129 @@ func TestQueryDocuments(t *testing.T) {
 		require.NoError(t, tx.Commit(ctx))
 	})
 }
+
+func TestGetDocuments(t *testing.T) {
+	t.Parallel()
+
+	ctx := testutil.Ctx(t)
+
+	pool := getPool(ctx, t)
+	databaseName := testutil.DatabaseName(t)
+	collectionName := testutil.CollectionName(t)
+	setupDatabase(ctx, t, pool, databaseName)
+
+	err := pool.InTransaction(ctx, func(tx pgx.Tx) error {
+		return CreateDatabaseIfNotExists(ctx, tx, databaseName)
+	})
+	require.NoError(t, err)
+
+	t.Run("one-document", func(t *testing.T) {
+		t.Parallel()
+
+		collection := collectionName + "-one"
+		expectedDoc := must.NotFail(types.NewDocument("_id", "foo", "id", "1"))
+
+		tx, err := pool.Begin(ctx)
+		require.NoError(t, err)
+		defer tx.Rollback(ctx)
+
+		require.NoError(t, InsertDocument(ctx, pool, databaseName, collection, expectedDoc))
+
+		sp := &SQLParam{DB: databaseName, Collection: collection}
+		it, err := pool.GetDocuments(ctx, tx, sp)
+		require.NoError(t, err)
+		require.NotNil(t, it)
+
+		iter, doc, err := it.Next()
+		assert.NoError(t, err)
+		assert.Equal(t, uint32(0), iter)
+		assert.Equal(t, expectedDoc, doc)
+
+		iter, doc, err = it.Next()
+		assert.Equal(t, iterator.ErrIteratorDone, err)
+		assert.Equal(t, uint32(0), iter)
+		assert.Nil(t, doc)
+
+		require.NoError(t, tx.Commit(ctx))
+	})
+
+	t.Run("cancel-context", func(t *testing.T) {
+		t.Parallel()
+
+		collection := collectionName + "-two"
+		expectedDocs := []*types.Document{
+			must.NotFail(types.NewDocument("_id", "bar", "id", "1")),
+			must.NotFail(types.NewDocument("_id", "foo", "id", "2")),
+		}
+
+		tx, err := pool.Begin(ctx)
+		require.NoError(t, err)
+		defer tx.Rollback(ctx)
+
+		require.NoError(t, InsertDocument(ctx, pool, databaseName, collection, expectedDocs[0]))
+		require.NoError(t, InsertDocument(ctx, pool, databaseName, collection, expectedDocs[1]))
+
+		ctxTest, cancel := context.WithCancel(ctx)
+		sp := &SQLParam{DB: databaseName, Collection: collection}
+		it, err := pool.GetDocuments(ctxTest, tx, sp)
+		require.NoError(t, err)
+		require.NotNil(t, it)
+
+		iter, doc, err := it.Next()
+		assert.NoError(t, err)
+		assert.Equal(t, uint32(0), iter)
+		assert.Equal(t, expectedDocs[0], doc)
+
+		cancel()
+		iter, doc, err = it.Next()
+		assert.EqualError(t, context.Canceled, err.Error())
```

Comment:
```suggestion
		assert.Equal(t, context.Canceled, err)
```

No need to compare strings